encodings/fastlanes/src/bitpacking/mod.rs

```diff
@@ -264,8 +264,17 @@ impl CanonicalVTable<BitPackedArray> for BitPackedEncoding {
         array: BitPackedArray,
         builder: &mut dyn ArrayBuilder,
     ) -> VortexResult<()> {
-        // TODO(joe): add specialised impl
-        builder.extend_from_array(array.into_array())
+        match_each_integer_ptype_with_unsigned_type!(array.ptype(), |$T, $UnsignedT| {
+            unpack_into::<$T, $UnsignedT, _, _>(
+                array,
+                builder
+                    .as_any_mut()
+                    .downcast_mut()
+                    .vortex_expect("bit packed array must canonicalize into a primitive array"),
+                |x| unsafe { std::mem::transmute(x) },
+                |x| unsafe { std::mem::transmute(x) },
```

Comment:
Some safety and explanation comment would be good here?

Reply:
Done.